sdk/keyvault/Azure.Security.KeyVault.Keys/tests/ManagedHsmCryptographyClientLiveTests.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Security.KeyVault.Keys.Cryptography;
+using NUnit.Framework;
+
+namespace Azure.Security.KeyVault.Keys.Tests
+{
+    [ClientTestFixture(KeyClientOptions.ServiceVersion.V7_2)]
+    public class ManagedHsmCryptographyClientLiveTests : CryptographyClientLiveTests
+    {
+        private static readonly IEnumerable<KeyOperation> s_aesKeyOps = new[]
+        {
+            KeyOperation.Encrypt,
+            KeyOperation.Decrypt,
+            KeyOperation.WrapKey,
+            KeyOperation.UnwrapKey,
+        };
+
+        public ManagedHsmCryptographyClientLiveTests(bool isAsync, KeyClientOptions.ServiceVersion serviceVersion)
+            : base(isAsync, serviceVersion, null /* RecordedTestMode.Record /* to re-record */)
+        {
+        }
+
+        public override Uri Uri =>
+            Uri.TryCreate(TestEnvironment.ManagedHsmUrl, UriKind.Absolute, out Uri uri)
+                ? uri
+                // If the AZURE_MANAGEDHSM_URL variable is not defined, we didn't provision one
+                // due to limitations: https://github.com/Azure/azure-sdk-for-net/issues/16531
+                : throw new IgnoreException($"Required variable 'AZURE_MANAGEDHSM_URL' is not defined");
```

Comment:
I like to put checks like these into the TestEnvironment itself but have no strong opinion here.

Reply:
I had considered something like that, but there are other places where I do (or maybe did - might've refactored it out, but could again) check if this is null to make some decisions, so throwing from a property wouldn't be ideal. Besides, I may not always want to simply ignore the tests but actually fail, so I'd rather leave the behavior defined by the test/fixture.